src/Bicep.Core/TypeSystem/TypePropertyFlags.cs

```diff
@@ -57,6 +57,11 @@ public enum TypePropertyFlags
         /// On non-required properties, this allows the property type to be treated as "<x> | null" (where <x> is the current property type)
         /// for the purposes of type checking the value assigned to the property.
         /// </summary>
-        AllowImplicitNull = 1 << 8
+        AllowImplicitNull = 1 << 8,
+
+        /// <summary>
+        /// Property that is not defined in Swagger, but still might be valid. We will show warning instead error.
```

Comment:
The behavior on property assignment looks good, but there are some inconsistencies in property access (see screenshot for details)

![image](https://user-images.githubusercontent.com/38542602/124981089-267a3900-e003-11eb-85bc-bc28aa24b213.png)

Reply:
Agreed. I'll fix it.

Reply:
@anthony-c-martin - should it also work for `existing` and in general for all top-level ones or only on those that were assigned? the second option complicates the logic a bit...

Reply:
> @anthony-c-martin - should it also work for `existing` and in general for all top-level ones or only on those that were assigned? the second option complicates the logic a bit...

I think the analysis on access should be totally independent of assignment - e.g. we show a warning (not an error) on a fallback property access regardless of whether it has been assigned to. That may simplify the implementation for `existing` a little bit.

I think for the implementation you could just copy the full set of fallback properties from declared -> assigned type, and avoid doing any narrowing on them.

Reply:
Fixed.